### PR TITLE
docs: MAPPO network

### LINKED ISSUE #652: [MAINTAIN] Jax MAPPO documentation
### Please describe what needs to be maintained?
Jax MAPPO needs documentation: everything within `mava/systems/jax/mappo`

### Describe the outcome you'd like
All classes / methods / functions should be clearly documented with explanations and descriptions of all args / returns

### PR DESCRIPTION
**Pull request by siddarthsingh1**
## What?
Document JAX Mava MAPPO 
## Why?
Jax mava systems need documentation.
## How?
Filled out docstrings and added comments.
## Extra
Closes #652

### CODE REVIEW

File: mava/systems/jax/mappo/networks.py [hunk 1]
```diff
@@ -119,7 +168,19 @@ def make_networks(
     critic_layer_sizes: Sequence[int] = (512, 512, 256),
     observation_network: Callable = utils.batch_concat,
 ) -> PPONetworks:
-    """TODO: Add description here."""
+    """Calls functions to make discrete or continuous network
```
Comment (OmaymaMahjoub) [inline mava/systems/jax/mappo/networks.py:171]: Add  Raises descriptions from these lines:
`raise NotImplementedError(
            "Continuous networks not implemented yet."
            + "See: https://github.com/deepmind/acme/blob/"
            + "master/acme/agents/jax/ppo/networks.py"
        )`

**Review by OmaymaMahjoub — CHANGES_REQUESTED** [review]
Thanks @siddarthsingh1 the documentation was really insightful and well written! One minor change, besides, I believe you also need to make a small description about `MAPPODefaultConfig` in config.py but it is optional!

File: mava/systems/jax/mappo/networks.py (hunk 1 above)
Comment (siddarthsingh1) [reply mava/systems/jax/mappo/networks.py:168]: Thanks @OmaymaMahjoub, it's been added.

**Review by OmaymaMahjoub — DISMISSED** [review]
Thanks @siddarthsingh1!

File: mava/systems/jax/mappo/components.py [hunk 2]
```diff
@@ -55,15 +55,23 @@ def __init__(
...
+            config : configuration for current algorithm
         """
         self.config = config
 
     def on_building_init_end(self, builder: SystemBuilder) -> None:
-        """[summary]"""
+        """Create extra specs after builder has been initialised
+
+        Args:
+            builder: builds system
```
Comment (AsadJeewa) [inline mava/systems/jax/mappo/components.py:69]: same here: in the components docs PR we have builder: SystemBuilder. I agree that descriptions are needed when the var name is not intuitive/ it is not obvious what the arg does

File: mava/systems/jax/mappo/networks.py [hunk 3]
```diff
@@ -49,7 +49,15 @@ def __init__(
         entropy: Optional[EntropyFn] = None,
         sample: Optional[networks_lib.SampleFn] = None,
     ) -> None:
-        """TODO: Add description here."""
+        """Initialises the PPO network Class.
+
+        Args:
+            network: feedforward network representing the agent policy function.
```
Comment (AsadJeewa) [inline mava/systems/jax/mappo/networks.py:55]: won't we use the same classes for recurrent PPO? Then this would not have to be feedforward

**Review by AsadJeewa — CHANGES_REQUESTED** [review]
Thanks, @siddarthsingh1. Few minor comments and questions. 

In general, the networks class has too many nested methods (`make_default_networks()` to `make_networks()` to `make_discrete_networks()` to `make_ppo_netrworks()`). This is not ideal and I think we can improve readability in another PR/ issue

File: mava/systems/jax/mappo/networks.py (hunk 3 above)
Comment (siddarthsingh1) [reply mava/systems/jax/mappo/networks.py]: Changed to  `neural network`

File: mava/systems/jax/mappo/components.py (hunk 2 above)
Comment (siddarthsingh1) [reply mava/systems/jax/mappo/components.py]: Changed to be more in line with the other docs.

File: mava/systems/jax/mappo/components.py [hunk 4]
```diff
@@ -55,15 +55,23 @@ def __init__(
         self,
         config: ExtrasLogProbSpecConfig = ExtrasLogProbSpecConfig(),
     ):
-        """_summary_
+        """Class that adds log probs to the extras spec
 
         Args:
-            config : _description_.
+            config : configuration for current algorithm
```
[the remark this answers is not in the document]
Comment (siddarthsingh1) [reply mava/systems/jax/mappo/components.py]: Changed to be more in line with the other docs.

**Review by AsadJeewa — APPROVED** [review]
Thanks @siddarthsingh1

**Review by OmaymaMahjoub — APPROVED** [review]
Looking good from my side! Thanks @siddarthsingh1

**Pull request merged**